src/utils/works.ts

```diff
@@ -7,16 +7,57 @@ import { getTagUrl } from "./tags";
 // correct type of page is passed to each method that extracts data.
 // Other than this, all pages are instances of CheerioAPI and can be used interchangeably.
 interface WorksPage extends CheerioAPI {
-  kind: "WorksPage";
+ kind: "WorksPage";
 }
+
+export interface WorkData {
+ workUrl: string;
+ workId: string;
+ chapterId: string | null;
+ collectionName: string | null;
+}
+
 const getWorksUrl = (tagName: string) => `${getTagUrl(tagName)}/works`;
 
 export const getWorksPage = async (tagName: string) => {
-  return cheerio.load(
-    (await axios.get<string>(getWorksUrl(tagName))).data
-  ) as WorksPage;
+ return cheerio.load(
+  (await axios.get<string>(getWorksUrl(tagName))).data
+ ) as WorksPage;
 };
 
 export const getTagId = ($worksPage: WorksPage) => {
-  return $worksPage(".rss")[0]?.attribs["href"].split("/")[2] || null;
+ return $worksPage(".rss")[0]?.attribs["href"].split("/")[2] || null;
+};
+
+interface WorkPage extends CheerioAPI {
+ kind: "WorkPage";
+}
+
+export const getWorkIdFromUrl = (url: string) => {
```

Comment:
What do you say about unifying these in a single `getWorkDataFromUrl` method?

Your return type will be:
```
{
  workId: string;
  chapterId?: string;
  collectionName?: string;
}
```

`?` indicates something might or might not be there. Generally these regexes operations are pretty quick, so there's no problem in just always running them and giving the requester complete data, even if they don't need all of it.